packages/kbn-config/src/deprecation/apply_deprecations.test.ts

```diff
@@ -116,6 +116,36 @@ describe('applyDeprecations', () => {
     expect(migrated).toEqual({ foo: 'bar', newname: 'renamed' });
   });
 
+  it('works for nested properties', () => {
```

Comment:
optional: It's not very descriptive. Let's document the problem you fix in the PR.